pkg/source/reconciler/common/finalizer.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package common
+
+import (
+	"context"
+
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+
+	kubeclient "knative.dev/pkg/client/injection/kube/client"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/reconciler"
+
+	"knative.dev/eventing-kafka/pkg/apis/sources/v1beta1"
+	"knative.dev/eventing-kafka/pkg/source/client"
+)
+
+const (
+	maxFinalizerAttempts = 10
+)
+
+// BoundedFinalizer limits the number of time a finalizer can run, due to too many errors.
+type BoundedFinalizer struct {
+	FinalizerAttempts map[string]int
```

Comment:
Fairly sure this map needs a mux guard as it might be accessed by parallel FinalizeKind calls.

Reply:
🙈 Thanks!